### PR TITLE
Updates for SensioFrameworkExtraBundle latest version(^5.2)

### LINKED ISSUE #3263: Update sensio/framework-extra-bundle to the latest version (v5.2.0)

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #3263

### CODE REVIEW

File: bundles/CoreBundle/Templating/LegacyTemplateGuesser.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+     * @var $container
+     */
+    private $container;
+
+    /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    public function __construct(KernelInterface $kernel, ContainerInterface $container, $controllerPatterns = [])
```
Comment (dpfaffenbauer) [inline bundles/CoreBundle/Templating/LegacyTemplateGuesser.php]: is it really necessary to pass in the container here? when you only need 'templating' from it?

Comment (brusch) [reply bundles/CoreBundle/Templating/LegacyTemplateGuesser.php]: @dvesh3 can you please fix that? Thanks :)

Comment (dvesh3) [reply bundles/CoreBundle/Templating/LegacyTemplateGuesser.php]: @brusch done :)

Comment (dpfaffenbauer) [reply bundles/CoreBundle/Templating/LegacyTemplateGuesser.php]: cool :), I guess the TemplateGuesser and and TemplateListener is for the auto rendering right?

Comment (dvesh3) [reply bundles/CoreBundle/Templating/LegacyTemplateGuesser.php]: yup, Guesser generates the tempate path and Listener is responsible for rendering it.

File: lib/Twig/Extension/HelpersExtension.php [hunk 2]
```diff
@@ -17,10 +17,12 @@
 
 namespace Pimcore\Twig\Extension;
 
+use Twig\Extension\AbstractExtension;
+
 /**
  * Simple helpers that do not need a dedicated extension
  */
-class HelpersExtension extends \Twig_Extension
+class HelpersExtension extends AbstractExtension
```
Comment (blankse) [inline lib/Twig/Extension/HelpersExtension.php]: Should not we also replace all deprecated \Twig_Test and \Twig_Function in all twig extensions?

File: lib/Routing/Loader/AnnotatedRouteControllerLoader.php [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 namespace Pimcore\Routing\Loader;
 
-use Sensio\Bundle\FrameworkExtraBundle\Routing\AnnotatedRouteControllerLoader as BaseAnnotatedRouteControllerLoader;
+use \Symfony\Bundle\FrameworkBundle\Routing\AnnotatedRouteControllerLoader as BaseAnnotatedRouteControllerLoader;
```
Comment (blankse) [inline lib/Routing/Loader/AnnotatedRouteControllerLoader.php]: The `\` at the beginning not necessary?

File: bundles/CoreBundle/EventListener/LegacyTemplateListener.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+class LegacyTemplateListener extends TemplateListener
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    public function __construct(TemplateGuesser $templateGuesser, \Twig_Environment $twig = null, ContainerInterface $container)
+    {
+        $this->container = $container;
```
Comment (blankse) [inline bundles/CoreBundle/EventListener/LegacyTemplateListener.php]: Can also replaced by DelegatingEngine $templateEngine?

File: lib/Twig/Extension/HelpersExtension.php (hunk 2 above)
Comment (dvesh3) [reply lib/Twig/Extension/HelpersExtension.php]: Sure these are deprecated? https://twig.symfony.com/doc/1.x/deprecated.html#tests https://twig.symfony.com/doc/1.x/deprecated.html#functions

File: lib/Routing/Loader/AnnotatedRouteControllerLoader.php (hunk 3 above)
Comment (dvesh3) [reply lib/Routing/Loader/AnnotatedRouteControllerLoader.php]: good catch ;)

File: bundles/CoreBundle/EventListener/LegacyTemplateListener.php (hunk 4 above)
Comment (dvesh3) [reply bundles/CoreBundle/EventListener/LegacyTemplateListener.php]: I tried but tests were failing...need to debug!

File: lib/Twig/Extension/HelpersExtension.php (hunk 2 above)
Comment (blankse) [reply lib/Twig/Extension/HelpersExtension.php]: In the 2.x doc it is only marked as final:
https://twig.symfony.com/doc/2.x/deprecated.html

In the code the `\Twig_Test` and `\Twig_Function` throw deprecation warnings.
`\Twig\TwigTest` and `\Twig\TwigFunction` doesn't throws deprecation warnings.

I think the 1.x doc is out of date.

Comment (blankse) [reply lib/Twig/Extension/HelpersExtension.php]: I think it cames from search and replace over the doc: https://github.com/twigphp/Twig/commit/3632a46e82fbb391b0fc72674bd8946e0b7c5bf7#diff-dd6d4937e8b99ee5a7e9deb418f67b30L74

Comment (dvesh3) [reply lib/Twig/Extension/HelpersExtension.php]: yes, AFAIK we are not overriding these classes ;)

Comment (blankse) [reply lib/Twig/Extension/HelpersExtension.php]: I create a PR for the misleading doc: https://github.com/twigphp/Twig/pull/3001 :)

**Pull request merged**